bahmnicore-omod/src/main/java/org/bahmni/module/bahmnicore/web/v1_0/controller/BahmniPatientProfileResource.java

```diff
@@ -246,13 +246,23 @@ private String getValueFromMap(Map<String, Object> jsonMap, String key) {
         Object value = jsonMap.get(key);
         return ObjectUtils.toString(value);
     }
-
+    
     private Relationship createRelationship(Person currentPerson, Map<String, Object> relationshipJson) {
-        Relationship relationship = new Relationship(currentPerson,
-                getPerson((Map<String, Object>) relationshipJson.get("personB")),
-                getRelationshipType((Map<String, Object>) relationshipJson.get("relationshipType")));
+        Relationship relationship;
+        String direction = (String) relationshipJson.get("direction");
+        
+        if (direction.equals("aToB")) {
+            relationship = new Relationship(currentPerson,
+                    getPerson((Map<String, Object>) relationshipJson.get("person")),
+                    getRelationshipType((Map<String, Object>) relationshipJson.get("relationshipType")));
+        } else {
```

Comment:
In general, I would do the opposite. The default should assume that we are in a `A_to_B` situation, not the other way.

Reply:
@Andu033 , this is still not in the right way. I would do the opposite of doing something specific only when the case is specificall `B_TO_A`. All other cases would be considered `A_TO_B`. Does that make sense?